Documentation/security/restrict-pod-access.rst

```diff
@@ -0,0 +1,52 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _security_restrict_pod_access:
+
+========================================
+Restricting privileged Cilium pod access
+========================================
+
+This page shows you how to restrict priveleged access to Cilium pods by limiting access from the Kubernetes API, specifically from `kubernetes exec pod`_.
+
+.. include:: gsg_requirements.rst
+
+Background
+----------
+
+While the Cilium agent no longer requires a specific set of privileges and capabilities, the Cilium agent requires specific privileges to perform essential system and network operations. 
+
+Cilium relies on Kubernetes and Containers to mount filesystems and the environment. Cilium doesn't perform any extra operations that could result in an unsafe volume mount. 
+
+Cilium needs kernel interfaces to properly configure the environment. Some kernel interfaces are part of the ``/proc`` filesystem, which includes host and machine configurations that can't be virtualized or namespaced.
+
+If ``pod exec`` operations aren't restricted, then remote `exec into pods`_ and containers defeats restrictions on Linux namespaces and capabilities.
+
+Entering a container that runs with execution (``X``) privileges like Cilium requires the ``CAP_SYS_ADMIN ``capability in your current user namespace and in the target Cilium user namespace. Because Cilium runs in the initial user namespace, it requires ``CAP_SYS_ADMIN`` capability, which provides essentially full root access.
```

Comment:
Entering a container that runs with (``X``) privileges like Cilium requires the ``CAP_SYS_ADMIN ``capability in your current user namespace and in the target Cilium user namespace. Since Cilium runs in the initial user namespace, then to enter its namespace the ``CAP_SYS_ADMIN`` capability is required, which is already a privileged access.

Reply:
@tixxdz 👋🏻 Some questions and feedback:
> Entering a container that runs with (X) privileges

What does `X` mean here? Is it in the chmod sense? It's better to explicitly state the meaning rather than assume a reader knows.

> Since Cilium runs

"Because" is [more accurate here](https://blog.apastyle.org/apastyle/2011/05/since-versus-because.html#:~:text=According%20to%20the%206th%20edition,is%20what%20is%20really%20meant.).

Reply:
> @tixxdz 👋🏻 Some questions and feedback:
> 
> > Entering a container that runs with (X) privileges
> 
> What does `X` mean here? Is it in the chmod sense? It's better to explicitly state the meaning rather than assume a reader knows.

Alright, we could simplify and connect with previous paragraph: The Linux kernel restricts joining other namespaces by default. To enter the Cilium container, the CAP_SYS_ADMIN capability is required in the current user namespace and in the Cilium user namespace which is the initial namespace. If these conditions are met, then this is already a privileged access.

Feel free to adapt.

> 
> > Since Cilium runs
> 
> "Because" is [more accurate here](https://blog.apastyle.org/apastyle/2011/05/since-versus-because.html#:~:text=According%20to%20the%206th%20edition,is%20what%20is%20really%20meant.).

Reply:
@tixxdz Thanks, that's helpful!